blocks/editable/index.js

```diff
@@ -68,18 +68,26 @@ export default class Editable extends Component {
 		this.onSelectionChange = this.onSelectionChange.bind( this );
 		this.maybePropagateUndo = this.maybePropagateUndo.bind( this );
 		this.onPastePostProcess = this.onPastePostProcess.bind( this );
+		this.onToolbarAddLink = this.onToolbarAddLink.bind( this );
+		this.onToolbarEditLink = this.onToolbarEditLink.bind( this );
+		this.onToolbarChangeLinkValue = this.onToolbarChangeLinkValue.bind( this );
 
 		this.state = {
 			formats: {},
-			bookmark: null,
 			empty: ! props.value || ! props.value.length,
+			toolbar: {
+				isAddingLink: false,
+				isEditingLink: false,
+				newLinkValue: '',
+			},
```

Comment:
I'm not super familiar with this code, but it seems like we're trying to force this component to be treated as a [controlled component](https://facebook.github.io/react/docs/forms.html#controlled-components), maintaining the value as it's changed? This seems like a good use case for an [uncontrolled component](https://facebook.github.io/react/docs/uncontrolled-components.html), meaning the link toolbar starts with an empty initial value when first shown, and only at the point that we decide we want to confirm a link being added do we trigger the `onAddLink` callback.